tmt/steps/provision/__init__.py

```diff
@@ -56,6 +56,10 @@ def load(self, extra_keys=None):
 
     def save(self, data=None):
         """ Save guest data to the workdir """
+        if not self.guests():
```

Comment:
I feel like this fixes the effect of the bug rather than its cause which I am not a big fan of. I feel like the bug is here: https://github.com/teemtee/tmt/blob/main/tmt/steps/provision/__init__.py#L121

To be exact, we shouldn't run `save()` if a `SpecificationError` is thrown, similarly to `SystemExit`, I'd say. Also we shouldn't add guest to the list of guests if it is None (which is in this case).

Moreover, it could make sense to add `required=True` to the `--guest` option, then this will be caught by click before actually doing anything.

Reply:
Hi. Good catch. I added required=True and disable save when SpecificationError exception is raised.

Reply:
One more small bug, if you setup a simple plan, e.g.

```yaml
summary: Basic smoke test
provision:
    how: connect
execute:
    script: tmt --help
```
and use `tmt run` (whole plan, including prepare/finish), then it still tracebacks. I think the problem here is that above your change, `None` is added to the list of guests here and then prepare/finish try to run stuff on top of it:

```python
                    if isinstance(plugin, ProvisionPlugin):
                        self._guests.append(plugin.guest())

```
I think we are missing a check if the guest has been provisioned (= is not None)

Reply:
Thanks. What about new change? :)